zilliqa/src/consensus.rs

```diff
@@ -588,7 +604,7 @@ impl Consensus {
         trace!("checking if block view {} is safe", block.view());
 
         // If the proposed block is safe, vote for it and advance to the next round.
-        if self.check_safe_block(&block)? {
+        if self.check_safe_block(&block, during_sync)? {
```

Comment:
So basically one of the issues is that when one of the nodes is behind, it needs to receive blocks and execute them.

The logic it needs to execute them (and all checks) is identical for the proposal logic, with the exception that in certain circumstances, the node's own view can be ahead of the block (but the block is still valid).

One way this can happen is if the node is the only one to get a propose message. It will increment its view after executing the propose message, but the rest of the network will be on a lower view and will eventually propose an alternate block.

So, if you are executing the block during sync, you can ignore specifically the view check.